### DISCUSSION TITLE
Apple's kerfuffle with "deleted" photos reappearing

### DISCUSSION THREAD

**Discussion by cfc62** [RhetTbull/osxphotos#1547] [General]
Anyone with thoughts about this?

I don't use iCloud Photos and I don't cull much so I haven't seen this happening on any of my devices, at least not that I've been able to detect.

**Comment by RhetTbull** [comment]
I'm not familiar with the issue -- do you have more details? I've never seen deleted photos re-appear.

**Reply by cfc62** [reply]
It's all the rage on Apple Rumors, Reddit and more...

https://www.macrumors.com/2024/05/15/ios-17-5-bug-deleted-photos-reappear/

https://www.reddit.com/r/iphone/comments/1csilpy/latest_ios_update_has_brought_back_some_pictures/

https://www.imore.com/iphone/this-weird-ios-175-bug-could-restore-years-old-deleted-iphone-photos-whether-you-want-it-to-or-not-and-theres-nothing-you-can-do-to-stop-it

There's more, but that's the gist of it... If you read the comments, people are generally pretty unhappy about this; this makes sense, if you deleted a compromising picture of you or someone else, you don't want it re-appearing down the road...

**Reply by cfc62** [reply]
And now The Verge: https://www.theverge.com/2024/5/15/24157284/apple-iphone-ios-17-5-update-deleted-photos-voicemails

**Reply by RhetTbull** [reply]
Interesting -- I hadn't seen this. If I could get my hands on a Photos library that this happened to I might be able to put in a filter for osxphotos to find these (but would need to see the full metadata from the database). osxphotos has some comparison tools (`osxphotos compare` and `osxphotos snap` & `osxphotos diff`) but these wouldn't help here. `osxphotos compare` requires another copy of the library to compare to. `snap` and `diff` would work to show changes but only at the raw SQL level in the database. You might be able to use `osxphotos query --added-in-last "1 week"` for example to show new photos. If I have time I will take a look at creating a small script that would create a snapshot of the library then allow you to see any photos added since the last snapshot. That would make it easier to find the undeleted photos.

**Comment by cfc62** [comment]
...and now iOS 17.5.1 is out to address the issue. 

Weird, I would have thought this was all cloud-side issues.

**Reply by oPromessa** [reply]
@cfc62   Do you know if it was ever clarified if the issue was:

- At the physical device level (not cleaned up/transfered/sold)
- at the physical device level which was cleaned up/transfered/sold
- Or did it spread out into the iCloud and other devices

Thanks.

**Reply by cfc62** [reply]
I sure don't. There's little clarity around this issue, that's for sure. 

There are some snippets floating out there talking about this was on-device, where deleted photos were "orphaned" on-device (specifically iOS) but then somehow re-found in 17.5.0, possibly as new code in 17.5.0 could see these orphaned photos and then recover them as new photos. Hence, the new iOS release. Interestingly, no corresponding MacOS release. 

I neither delete photos or use iCloud Photos so I don't have a lot of experience there, but I will say the approach Apple uses in just "moving" photos to a holding pen where they will be permanently deleted in 30 days has caught me out before. 

Also oddly enough I just experienced previously deleted emails popping back up on a cloud email provider (not Apple related), so I do know that can happen. 

To me, this is yet another example of where's Apple's lack of documentation and communication is not helpful.

**Reply by cfc62** [reply]
Found this, which seems to make sense:

https://www.reddit.com/r/ios/comments/1cwgljj/regarding_the_ios_175_photo_glitch/

**Reply by cfc62** [reply]
And now there's a tvOS release for the same issue. I have ATVs, but I can't think of any way besides iCloud Photos to get images in there. So maybe there's a cloud component after all.

**Comment by cfc62** [comment]
https://9to5mac.com/2024/05/23/apple-deleted-photos-resurfacing-explanation/

It's not much, but Apple has at least said something (although unofficially).

**Reply by cfc62** [reply]
This is better: https://www.synacktiv.com/en/publications/inside-the-ios-bug-that-made-deleted-photos-reappear

**Reply by RhetTbull** [reply]
Thanks! Interesting bug.
